src/Compiler/Service/ItemKey.fs

```diff
@@ -278,20 +362,39 @@ and [<Sealed>] ItemKeyStoreBuilder() =
         | _ -> ()
 
     and writeTypar (isStandalone: bool) (typar: Typar) =
+        debug.WriteTypar isStandalone typar
+
         match typar.Solution with
         | Some ty -> writeType isStandalone ty
         | _ ->
             if isStandalone then
                 writeInt64 typar.Stamp
 
     let writeValRef (vref: ValRef) =
+        debug.WriteValRef vref
+
         match vref.MemberInfo with
         | Some memberInfo ->
             writeString ItemKeyTags.itemValueMember
-            writeEntityRef memberInfo.ApparentEnclosingEntity
+
+            match vref.IsOverrideOrExplicitImpl, vref.MemberInfo with
+            | true,
+              Some {
+                       ImplementedSlotSigs = slotSig :: _tail
+                   } -> slotSig.DeclaringType |> writeType false
+            | _ -> writeEntityRef memberInfo.ApparentEnclosingEntity
+
             writeString vref.LogicalName
             writeString ItemKeyTags.parameters
-            writeType false vref.Type
+
+            if vref.IsInstanceMember && isFunTy tcGlobals vref.Type then
+                // In case of an instance member, we will skip the type of "this" because it will differ
+                // between the definition and overrides. Also it's not needed to uniquely identify the reference.
+                destFunTy tcGlobals vref.Type |> snd
```

Comment:
Probably should be using `tryDestFunTy` here instead, too many times it was blowing up.

Reply:
Yeah that sounds better, I just didn't notice it.